### DISCUSSION TITLE
Apply augmentation stack with different parameters to multiple images

### DISCUSSION THREAD

**Discussion by mikel-brostrom** [albumentations-team/albumentations#1406] [Ideas]
I have the following piece of code:

```python
import random

import cv2
from matplotlib import pyplot as plt

import albumentations as A

from utils.augmentations import Mosaic


def visualize(image):
    plt.figure(figsize=(10, 10))
    plt.axis('off')
    plt.imsave('blub.jpg',image)
    

image = cv2.imread(./images/train2017/000000000139.jpg')
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
image1 = cv2.imread('./images/train2017/000000000285.jpg')
image1 = cv2.cvtColor(image1, cv2.COLOR_BGR2RGB)
image2 = cv2.imread(./images/train2017/000000000632.jpg')
image2 = cv2.cvtColor(image2, cv2.COLOR_BGR2RGB)
image3 = cv2.imread('./images/train2017/000000000724.jpg')
image3 = cv2.cvtColor(image3, cv2.COLOR_BGR2RGB)

imgsz = 416
transform = A.Compose(
    [
        A.augmentations.transforms.HueSaturationValue(p=0.1),  # hsv change default: [20, 30, 20]
        A.augmentations.transforms.ColorJitter(p=0.1),
        A.augmentations.transforms.Sharpen(p=0.1),
        A.augmentations.transforms.ToGray(p=0.1),
        Mosaic(
            height=int(2.5*imgsz),
            width=int(2.5*imgsz),
            replace=False, p=1.0,
            fill_value=114,
            bboxes_format='coco'
        ),
    ],
    additional_targets={'image1': 'image', 'image2': 'image', 'image3': 'image'}
)

transformed = transform(image=image, image1=image1, image2=image2, image3=image3)

visualize(transformed['image'])
```

Which gives me:

![blub1](https://user-images.githubusercontent.com/18719680/220465542-3f870a82-8c9d-4abe-b250-c033117c74dc.jpg)
![blub](https://user-images.githubusercontent.com/18719680/220465547-10d0dab8-751c-455c-9a13-1f7a782c25a1.jpg)

for two different runs. It would be nice if the augmentations could be applied to the input images with a different set of parameters to increase variability on the training data :rocket:. At the moment, the default behavior is that the augmentations are applied to all images with the same parameters. Would it be possible to implement this?

**Comment by Dipet** [comment]
> At the moment, the default behavior is that the augmentations are applied to all images with the same parameters. Would it be possible to implement this?

What a problem to call transforms twice for this reason?

**Reply by mikel-brostrom** [reply]
I am working on passing multiple images to a custom Mosiac implementation that takes in 4 images. It would simplify things if I wouldn't need to call my augmentation stack for each image.

**Reply by mikel-brostrom** [reply]
Maybe I should add it as an enhancement proposal?

**Reply by Dipet** [reply]
I don't think that it is a good idea. If you need different applies of transform pipeline you must call pipeline multiple times.

**Reply by mikel-brostrom** [reply]
Alright
